src/plugins/explore/public/utils/trace_field_validation.ts

```diff
@@ -0,0 +1,214 @@
+} from './trace_field_constants';
+
+export const START_TIME_FIELD_PATHS = [
+  'startTimeUnixNano',
+  'startTime',
+  '_source.startTimeUnixNano',
+  '_source.startTime',
+  'fields.startTimeUnixNano',
+  'fields.startTime',
+] as const;
+
+export const END_TIME_FIELD_PATHS = [
+  'endTimeUnixNano',
+  'endTime',
+  '_source.endTimeUnixNano',
+  '_source.endTime',
+  'fields.endTimeUnixNano',
+  'fields.endTime',
+] as const;
+
+export const PARENT_SPAN_ID_FIELD_PATHS = [
+  'parentSpanId',
+  'parent_span_id',
+  'parentSpanID',
+  'parent_span_ID',
+  '_source.parentSpanId',
+  '_source.parent_span_id',
+  'fields.parentSpanId',
+  'fields.parent_span_id',
+] as const;
+
+const getNestedValue = (obj: unknown, path: string): unknown => {
+  return path.split('.').reduce((current: unknown, key: string) => {
+    return current && typeof current === 'object' && current !== null
+      ? (current as Record<string, unknown>)[key]
+      : undefined;
+  }, obj);
+};
+
+export const extractFieldFromRowData = (
+  rowData: OpenSearchSearchHit<Record<string, unknown>>,
+  fields: readonly string[]
+): string => {
+  if (!rowData) return '';
+
+  for (const field of fields) {
+    const value = getNestedValue(rowData, field);
+    if (
+      value !== undefined &&
+      value !== null &&
+      (typeof value === 'string' || typeof value === 'number')
+    ) {
+      return String(value);
+    }
+  }
+  return '';
+};
+
+export const extractServiceNameFromRowData = (
+  rowData: OpenSearchSearchHit<Record<string, unknown>>
+): string => {
+  if (!rowData) return '';
+
+  const resourceServiceName = getNestedValue(rowData, 'resource.attributes.service.name');
+  if (resourceServiceName && typeof resourceServiceName === 'string') {
+    return resourceServiceName;
+  }
+
+  const resourceServiceNameAlt = getNestedValue(rowData, 'resource.attributes.service.name');
+  if (resourceServiceNameAlt && typeof resourceServiceNameAlt === 'string') {
+    return resourceServiceNameAlt;
+  }
+
+  const sourceServiceName = getNestedValue(rowData, '_source.serviceName');
+  if (sourceServiceName && typeof sourceServiceName === 'string') {
+    return sourceServiceName;
+  }
+
+  const serviceName = getNestedValue(rowData, 'serviceName');
+  if (serviceName && typeof serviceName === 'string') {
+    return serviceName;
+  }
+
+  const sourceName = getNestedValue(rowData, '_source.name');
+  if (sourceName && typeof sourceName === 'string') {
+    return sourceName;
+  }
+
+  const name = getNestedValue(rowData, 'name');
+  if (name && typeof name === 'string') {
+    return name;
+  }
+
+  return '';
+};
+
+export const validateRequiredTraceFields = (
+  rowData: OpenSearchSearchHit<Record<string, unknown>>
+): {
+  isValid: boolean;
+  missingFields: string[];
+  presentFields: string[];
+} => {
+  const missingFields: string[] = [];
+  const presentFields: string[] = [];
+
+  const spanId = extractFieldFromRowData(rowData, SPAN_ID_FIELD_PATHS);
+  if (spanId) {
```

Comment:
I've noticed that there is a repetitive pattern here with extract -> check -> push. We could try and make this a bit more maintainable and readable. 

(Completely optional suggestion):
We have an array of each of these fields we are checking for - this array could contain the name, the path we expect to find it at (to handle nested values), anything else needed.
Then, we have this function work to perform validation after on that array. 

A potential downside of doing my suggestion is that each field we check for is abstract enough that all of the various options inside of the array that we put down could grow to be larger than just doing this manually (which could be why you've done it this way). I haven't fully checked but take a look and see if this could help.